jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -1865,47 +2136,109 @@ public VarHandle unreflectVarHandle(Field field) throws IllegalAccessException {
 			 * as it is not set up for lookup objects by default.
 			 */
 			int fullAccessMode = FULL_ACCESS_MASK | MODULE | UNCONDITIONAL;
+			
+			switch(dropMode) {
+			case PUBLIC:
+			case MODULE:
+			case PACKAGE:
+			case PRIVATE:
+			case PROTECTED:
+			case UNCONDITIONAL:
+				/* dropMode is OK */
+				break;
+			default:
+				/*[MSG "K065R", "The requested lookup mode: 0x{0} is not one of the existing access modes: 0x{1}"]*/
+				throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K065R", Integer.toHexString(dropMode), Integer.toHexString(fullAccessMode))); //$NON-NLS-1$
+			}
 
+			/*[IF Java14]*/
+			/* The lookup object has to discard the protected access by default */
+			int newAccessMode = accessMode & ~PROTECTED;
+			/*[ELSE]*/
 			/* The lookup object has to discard the protected and unconditional access by default */
 			int newAccessMode = accessMode & ~(PROTECTED | UNCONDITIONAL);
+			/*[ENDIF] Java14*/
 			
-			switch (dropMode) {
+			/* The access mode to be dropped must exist in the current access mode;
+			 * otherwise, the new access mode remains unchanged.
+			 */
+			switch (dropMode & newAccessMode) {
 			case PUBLIC:
 				newAccessMode = NO_ACCESS;
 				break;
-			case MODULE:
-				newAccessMode &= ~(MODULE | PACKAGE | PRIVATE);
-				break;
 			case PACKAGE:
 				newAccessMode &= ~(PACKAGE | PRIVATE);
 				break;
 			case PRIVATE:
 				newAccessMode &= ~PRIVATE;
 				break;
-			case PROTECTED:
 			case UNCONDITIONAL:
+				/*[IF Java14]*/
+				newAccessMode = NO_ACCESS;
+				/*[ENDIF] Java14*/
 				break;
 			default:
-				/*[MSG "K065R", "The requested lookup mode: 0x{0} is not one of the existing access modes: 0x{1}"]*/
-				throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K065R", Integer.toHexString(dropMode), Integer.toHexString(fullAccessMode))); //$NON-NLS-1$
+				/* no change in the access mode */
 			}
 			
-			return new Lookup(accessClass, newAccessMode); 
+			/* The exception is MODULE in which case all access bits involved must be dropped
```

Comment:
Why is this an exception?  I don't see anything about this in the "Access modes" table showing dropLookupMode behaviour in https://download.java.net/java/early_access/jdk14/docs/api/java.base/java/lang/invoke/MethodHandles.Lookup.html#access-modes

It seems like this should apply to all dropped modes & the change to `switch (dropMode & newAccessMode)` is unnecessary

Reply:
As mentioned at https://github.com/eclipse/openj9/issues/8571#issuecomment-594864602,
```
  public void testDropLookupMode() throws Exception {
        Lookup lookup = MethodHandles.privateLookupIn(m5.type1, m4.lookup);
        assertTrue((lookup.lookupModes() & MODULE) == 0); <--- MODULE doesn't exist
        ...
        Lookup lookup3 = lookup.dropLookupMode(MODULE);
---> assertTrue(lookup3.lookupModes() == (lookup.lookupModes() & ~(PROTECTED|PRIVATE|PACKAGE)));
```

Here's the printing message for the test above
```
[TestNG] Running:
  java/lang/invoke/modules/Driver1.java

config jdk.test.ModuleAccessTest.setup(): success
----- testDropLookupMode --------
m4.lookup.lookupModes() = 31
m4.lookup.lookupClass() = class d1.D1
m4.lookup.lookupClass().getModule() = module m4
m5.type1.getModule() = module m5

lookup = e1.E1/d1.D1
lookup.lookupClass = class e1.E1
lookup.lookupClass().getModule() = module m5
lookup.lookupModes() = 15 <--------------------- 0xF or 1111 <---- MODULE doesn't exist
lookup.previousLookupClass() = class d1.D1
lookup.lookupModes() & ~(PROTECTED|PRIVATE|PACKAGE) = 1
lookup3.lookupModes() = 1 <------------------------------ PUBLIC is left eventually
test jdk.test.ModuleAccessTest.testDropLookupMode(): failure
```
Based on the weird result, the generated lookup  doesn't hold the MODULE bit, in which case it shouldn't drop all `PROTECTED|PRIVATE|PACKAGE` as if the MODULE bit exists if the previous clarification from Oracle (mode to be dropped must exist in the access mode) is reasonable; Otherwise, Oracle should bring up with another explanation as to how this happens and why.